roles/acm/utils/tasks/get-credentials.yml

```
@@ -15,8 +15,8 @@
 
 - name: Get kubeadmin credentials for HCP
   when:
-    - kubeconfig_secret.resources[0].metadata.labels.keys() | list is search('hypershift') or
-      kubeconfig_secret.resources[0].metadata.annotations.keys() | default({}) | list is search('hypershift')
+    - (kubeconfig_secret.resources[0].metadata.labels | default({}) | dict2items | map(attribute='key') | join(' ') is search('hypershift')) or
+      (kubeconfig_secret.resources[0].metadata.annotations | default({}) | dict2items | map(attribute='key') | join(' ') is search('hypershift'))
```

Wondering if this would help:
```suggestion
    - '"hypershift" in (kubeconfig_secret.resources[0].metadata.labels.keys() | default({}) | list)
    + (kubeconfig_secret.resources[0].metadata.annotations.keys() | default({}) | list)'
```

---

Thanks @tonyskapunk as the patch is now, it is working fine for HCP and non-HCP deployments.